pkg/pool-manager/pool.go

```diff
@@ -176,3 +177,24 @@ func getNextMac(currentMac net.HardwareAddr) net.HardwareAddr {
 
 	return currentMac
 }
+
+// Checks if the namespace of an instance is opted in for kubemacpool
+func (p *PoolManager) IsInstanceOptedIn(namespaceName string, labelSelector string) (error, bool) {
+
```

Comment:
Let me do some nitpicking. There is no need for an empty line after you open the function. The first line is already visually divided by the identation.